### ISSUE TITLE
Interface missing superclass fields

### ISSUE THREAD

**Issue by jshearer** [nestjs/graphql#1136]
## I'm submitting a...

<pre><code>
[ ] Regression 
[x] Bug report
[x] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead post your question on Stack Overflow.
</code></pre>

## Current behavior

It seems that when I define an interface on a class that extends a base class, the fields defined on that base class do not get propagated properly. So for example:

```typescript
// BaseEntity.ts

export abstract class BaseEntity {
    @Field(_type => ID)
    global_id: string;

    @Field()
    created_at: Date;

    @Field()
    updated_at: Date;
}
```

```typescript
// TaskInterface.ts

@InterfaceType()
export class Task extends BaseEntity {
    @Field(_type => String, { nullable: true })
    comments: string;
  
    // Shortened for legibility
}
```

```typescript
// InterfaceImplementation.ts

@ObjectType({ implements: [Task] })
export class Report extends Task {
    @Field(_type => [ReportParameter], { nullable: true })
    parameters: Array<ReportParameter>;
}
```

The resulting schema is:

```graphql
interface Task {
  comments: String
}

type Report implements Task {
  comments: String
  parameters: [ReportParameter!]
}
```

Critically, `global_id`, `created_at`, and `updated_at` are missing. Is this not supported? Or am I using the library incorrectly?

## Environment

<pre><code>
Nest version: 7.4.4

 
For Tooling issues:
- Node version: v12.17.0  
- Platform:  Mac, Linux

</code></pre>

**Comment by kamilmysliwiec** [comment]
`BaseEntity` isn't annotated with any decorator. Please, try using the `@InterfaceType({ isAbstract: true })` on top of this class

**Comment by jshearer** [comment]
@kamilmysliwiec, thanks for the advice. I tried that, but am now getting errors:

```typescript
@InterfaceType({ isAbstract: true })
export abstract class BaseEntity {
    @Field(_type => ID)
    global_id = v4();

    @Field()
    created_at: Date;

    @Field()
    updated_at: Date;
}

@InterfaceType()
export class Task extends BaseEntity {
    @Field(_type => String, { nullable: true })
    comments: string;
}

@ObjectType({ implements: [Task] })
export class Report extends Task {
    @Field(_type => [ReportParameter], { nullable: true })
    parameters: Array<ReportParameter>;
}
```
=>
```
GraphQLError [Object]: Interface field Task.global_id expected but Report does not provide it.
GraphQLError [Object]: Interface field Task.created_at expected but Report does not provide it.
GraphQLError [Object]: Interface field Task.updated_at expected but Report does not provide it.
```

It's almost as if the `@InterfaceType` on `Task` needs a hint that it should really also include the fields from its parent... thoughts?

This works if I do

```typescript
@ObjectType({ implements: [Task, BaseEntity] })
export class Report extends Task {
    ...
}
```
But that has the consequence of emitting another schema in the SDL, which I don't want -- I want the abstract behaviour where the fields are just included automatically.

**Comment by jshearer** [comment]
Well I'll be damned, reading the source actually worked. Turns out I needed to decorate Task with `@ObjectType({isAbstract: true})` in addition to `@InterfaceType()` in order for this to work.

Including fields from parents is supported for interfaces, as shown here: https://github.com/nestjs/graphql/blob/c121c5a923ce55af709337652de735f252b83ee6/lib/schema-builder/factories/object-type-definition.factory.ts#L80

But it seems that only parents that are `@ObjectTypes` are checked: https://github.com/nestjs/graphql/blob/c121c5a923ce55af709337652de735f252b83ee6/lib/schema-builder/factories/object-type-definition.factory.ts#L41-L46

Not sure if this is a bug or not, but at least from my perspective it seems as if this should "just work" without having to mark `Task` as an abstract `ObjectType` as well as an interface.
